client/ayon_core/plugins/publish/validate_file_saved.py

```diff
@@ -1,17 +1,55 @@
+import inspect
+
 import pyblish.api
 
 from ayon_core.pipeline.publish import PublishValidationError
+from ayon_core.tools.utils.host_tools import show_workfiles
+from ayon_core.pipeline.context_tools import version_up_current_workfile
+
+
+class SaveByVersionUpAction(pyblish.api.Action):
+    """Save Workfile."""
+    label = "Save Workfile"
+    on = "failed"
+    icon = "save"
+
+    def process(self, context, plugin):
+        version_up_current_workfile()
+
+
+class ShowWorkfilesAction(pyblish.api.Action):
+    """Save Workfile."""
+    label = "Show Workfiles Tool..."
+    on = "failed"
+    icon = "files-o"
+
+    def process(self, context, plugin):
+        show_workfiles()
 
 
 class ValidateCurrentSaveFile(pyblish.api.ContextPlugin):
```

Comment:
Should we add this into some other hosts which they need to have saved workfile for publish? (e.g. Substance Painter, 3dsMax etc.)

Reply:
This is in `ayon-core`. If it's valid for those hosts then we can add the relevant host to the `hosts = [x, y, z]` entries on this plug-in. Correct.

Reply:
Should I resolve for now - or do I need to add more hosts? Any chance you can do a test run with Substance Painter and Max @moonyuet ?

Reply:
1. Added substancepainter and max. 
2. Made a matching substancepainter PR to improve: https://github.com/ynput/ayon-substance-painter/pull/7
3. For Max we may want to remove Max's own "saved" validation in some time to allow this core one to be the nice and shiny one. For now, with this PR you might get two invalidations for unsaved files - one nice one, one legacy one from max itself.